### DISCUSSION TITLE
Configure the Counter & GPIO Sensor | Seeed Studio Wiki

### DISCUSSION THREAD

**Discussion by giscus** [Seeed-Studio/wiki-documents#894] [Q&A]
# Configure the Counter & GPIO Sensor | Seeed Studio Wiki

Configure the Counter & GPIO Sensor

https://wiki.seeedstudio.com/Sensor/SenseCAP/SenseCAP_Data_Logger/tutorial/How_to_Configure_the_Counter&GPIO_Sensor_for_S2100_Data_Logger/

**Comment by iothingcloud** [comment]
I followed the guide precisely, however no matter what i tried I could not get this to work correctly with the RG15 rain sensor and S2100 running 3.00 firmware (nor any firmware for that matter)

5V output: when setting to 5V it does not measure 5V between 5V and GND terminals, I assume it's not turning on correctly. Should it be enabled as soon as I save the settings over Bluetooth? Same for 3V output, does nothing useful. I therefore cannot power the sensor using the logger device. What is the point of warmup time if the rain sensor needs to be powered all the time? When powering my sensor separately on my home power supply it works, so the sensor is not faulty.

When I read the logger settings over bluetooth, "pull high" and "digital filter" are never simultaneously enabled. If I enable both, save them read the setting back, the "pull high" is now "pull low".

When enabling 'pull high', i never observe 5V on the IO pin, thus it does not measure the output from the sensor.

**Comment by moonlight0551** [comment]
@edak I found the same no 5V measured but it does work.(don't quite know why) I configured in the SenseCAP app on my phone using the settings in the sensecap document. Had trouble getting it to register on the TTN as I'm in Australia using AU915 Sub band 2. Had to select Network --> Other  and then enter the frequency range and then you get an option to enter the subband.. Made sure the EUI , AppKey etc all match and it joined successfully. It wouldn't join if I selected the Network --->The Thngs Network.. Once joined I could see a measurement of zero. Then I put the dome of the sensor under a tap with trickling water and the green led in the sensor started to flash(bright green). On the next update TTN there was a measurement recorded. I thought I had a dud data logger unit when I could need see that 5 volts and I changed it to another I had and same thing.  Another comment is that they seem to chew through batteries(maybe the sleep function doesn't work correctly) and have had to replace the battery every 6 - 9 months. Well hope this helps you or others struggling with this sensor. Cheers Dave

**Reply by Zeke5579** [reply]
Can you send me the model number of the sensor you are using? And the S2100 setup page. Maybe you are uploading too much information or the sensor is using too much power to get data at one time.

**Reply by moonlight0551** [reply]
The model of the sensor is the RG-15. I am uploading data to TTN every 30 minutes. Mine is working fine. I set it up using this document https://files.seeedstudio.com/products/SenseCAP/S2100/SenseCAP%20S2100%20LoRaWAN%20Data%20Logger%20User%20Guide.pdf

**Reply by iothingcloud** [reply]
The device was faulty and was replaced. 
Rg15 sending data every 6 hours now and has been working just fine for about a year now. 

When working with an analog sensor (the stainless steel depth probe from seed) the thing chews batteries like nothing else. Already replace battery twice in a year, when only taking one level reading every hour. That's a different thread and I had no response on this.

**Reply by Zeke5579** [reply]
Sensors using analog circuits usually consume more power, you can expand the time interval between sensors acquisitions and change the power supply mode to periodic.

**Reply by iothingcloud** [reply]
Hi Zeke, 

I have one of these sensors.
https://www.seeedstudio.com/Liquid-Level-Sensor-p-4619.html

My s2100 is configured for periodic 5V power to be applied to the sensor once per hour for 300ms.  Given the sensor is supposedly powered for 7.2s per day (total of 0.36h in 6 months), I have to conclude that the s2100 power circuit is problematic or the sensor is consuming stupid amount of power that the s2100 couldn't provide (multiple amps).

What would you say here?

**Reply by Zeke5579** [reply]
After my rough calculation, the power consumption of the sensor is 0.5w and the power consumption of the signal transmission is about 0.3w. Your battery is really draining too fast, we suggest you to seek more information and after-sales from sensecap@seeed.cc
